use dformat from dokuwiki by default

--- icalevents/conf/default.php~	2010-04-26 15:48:52.000000000 +0300
+++ icalevents/conf/default.php	2010-12-05 22:15:22.486210482 +0200
@@ -3,9 +3,7 @@
 
 # Date format that is used to display the from and to values
 # If you leave this empty '', then the default dformat from /conf/dokuwiki.php will be used.
-$conf['dformat'] = '%d.%m.%Y';
+$conf['dformat'] = '';
 
 # should the end dates for each event be shown?
 $conf['showEndDates'] = 0;
-
-?>
\ No newline at end of file
