--- iCalEvents/syntax.php~	2010-12-05 22:32:32.000000000 +0200
+++ iCalEvents/syntax.php	2010-12-05 22:32:57.535271298 +0200
@@ -132,7 +132,10 @@
      */
     function _parseIcs($url, $from, $previewSec) {
 	    global $conf;
-	
+
+        // must reset error in case we have multiple calendars on page
+        $this->error = false;
+
         $http    = new DokuHTTPClient();
         if (!$http->get($url)) {
           $this->error = "Could not get '$url': ".$http->status;
